lib.php

```diff
@@ -33,8 +33,67 @@
  */
 function report_roster_extend_navigation_course($navigation, $course, $context) {
     if (has_capability('report/roster:view', $context)) {
+        // If the setting is enabled, put a link to the report in the flat navigation.
+        if (get_config('report_roster', 'flatnav')) {
+            add_to_flatnav();
+        }
+
         $url = new moodle_url('/report/roster/index.php', array('id' => $course->id));
         $navigation->add(get_string('pluginname', 'report_roster'), $url,
                 navigation_node::TYPE_SETTING, null, null, new pix_icon('i/report', ''));
     }
 }
+
+/**
+ * Adds the Roster link to the flat course navigation.
+ *
+ * @return void
+ */
+function add_to_flatnav() {
```

Comment:
Is `add_to_flatnav` a core concept? Should this be named `report_roster_add_to_flatnav`?

Reply:
Good catch. Fixed.